code/modules/projectiles/projectile/magic.dm

```diff
@@ -430,6 +426,7 @@
 
 /obj/structure/closet/decay/proc/magicly_lock()
 	if(!welded)
+		addtimer(CALLBACK(src, .proc/decay), 15 SECONDS)
```

Comment:
is there a point for auto_destroy now if they decay automatically?

Reply:
Reason for why i did it was cause i felt that 5 minutes at least for the lockers that did not catch anything is too long trough i give in this is a bad way to implement it i will look into making sure only one timer runs at a time (so that there is not a totally obsolete timer that will not run out in time anyway.

Reply:
I will just use a switch expression to choose what timers to add

Reply:
Ok so much about using a switch or if in initialize sadly i still had to make the proc call before the check